### DISCUSSION TITLE
Is it possible to change PodCIDR to add IPv6?

### DISCUSSION THREAD

**Discussion by dedene** [siderolabs/talos#7015] [Q&A]
Currently, our clusters are running IPv4 with Cilium. I'm trying to enable IPv6 without destroying and rebuilding the cluster: just adding an IPv6 range to the podSubnets and reinstalling the node (using `talosctl upgrade --image ghcr.io/siderolabs/installer:v1.3.6`) does not seem to change the PodCIDRs reported when describing the nodes. 

Therefore I'm wondering: is it possible to change the PodCIDR after the cluster has been bootstrapped? Or is this something that is not possible at all?

**Comment by smira** [comment]
PodCIDRs in the machine config are applied automatically without a reboot, but as you're using Cilium, you need to make same change to the Cilium manifests - please refer to Cilium docs on how to configure dual-stack networking.

**Comment by dedene** [comment]
Thanks for your answer! The issue will probably be somewhere with the cloud controller manager I'm using. The IPv6 CIDR does not get set to the node and next when configuring Cilium it detects there is no IPv6 CIDR present so it throws `required IPv6 PodCIDR not available` (from [here](https://github.com/cilium/cilium/blob/master/pkg/k8s/init.go#L115)). I'll let it know if I find the root cause.

**Reply by dedene** [reply]
I indeed forgot to enable dualstack on the cloud controller manager. Now IPv6 seems to be working fine, thanks a lot!
